Add usbg_udc structure

kopasiak opened the pull request:
Dear Matt,

This series adds usbg_udc structure and fixes some typos introduced
in previous patches.

Before this series UDC didn't have its own structure. Gadget enable
took const char \* as parameter. Such approach is inconsisten with
rest of API.

This series introduce structure usbg_udc as abstraction of UDC.
This makes API more consistent and some functions (like enable
gadget) are now shorter and easier to read. Moreover usage of
such structure allows to implement usbg_get_udc_gadget() which
provides possibility to check which gadget is enabled on selected UDC.

In addition this series fix a lot of typos introduced in previous
patches. Now doxygen generates documentation without errors.

kopasiak committed:
libusbg: Fix off-by-one error.

Signed-off-by: Krzysztof Opasiak <k.opasiak@samsung.com>

kopasiak committed:
libusbg: Always add '\0' at end of string

strncpy() may not append trailing '\0' so let's append
it always at end of string to avoid getting into troubles.

Signed-off-by: Krzysztof Opasiak <k.opasiak@samsung.com>

kopasiak committed:
libusbg: doc: Fix typos in doxygen documentation

Signed-off-by: Krzysztof Opasiak <k.opasiak@samsung.com>

kopasiak committed:
libusbg: Replace usbg_get_udcs() with loop

Library allows to iterate over each gadget using loop.
To be consistent with this convention add similar loop
for UDCs.

Signed-off-by: Krzysztof Opasiak <k.opasiak@samsung.com>

kopasiak committed:
libusbg: Refresh gadget UDC while each usbg_get_gadget_udc()

Kernel may decide to detach our gadget in some cases.
A good example is when FFS daemon received a SIGSEGV
and all descriptors has been closed.

To avoid returning false informations to user we have
to refresh UDC attribute before returning cached pointer
to usbg_udc.

Signed-off-by: Krzysztof Opasiak <k.opasiak@samsung.com>

kopasiak committed:
libusbg: Allow to get gadget enabled on given UDC

Add usbg_get_udc_gadget() which allows user to learn
which gadget is attached to selected UDC.

Signed-off-by: Krzysztof Opasiak <k.opasiak@samsung.com>

kopasiak committed:
libusbg: Add example how to use usbg_udc structure

This example shows how to learn what udcs are available
in system and also how to find out what gadgets are
enabled on them.

Signed-off-by: Krzysztof Opasiak <k.opasiak@samsung.com>

philippedeswert committed:
libusbg: Return USBG_ERROR_PATH_TOO_LONG error when it happens

Signed-off-by: Philippe De Swert <philippe.deswert@jollamobile.com>
Acked-by: Krzysztof Opasiak <k.opasiak@samsung.com>
Signed-off-by: Krzysztof Opasiak <k.opasiak@samsung.com>

Update commit message.

Signed-off-by: Krzysztof Opasiak <k.opasiak@samsung.com>

philippedeswert commented:
I think I found some small issues and made fixes for them. They can be found here : https://github.com/kopasiak/libusbg/pull/1

kopasiak commented:
I have put some comments about them. Please see my comments in : https://github.com/kopasiak/libusbg/pull/1

kopasiak committed:
libusbg: Simplify getting names form library structures

Signed-off-by: Krzysztof Opasiak <k.opasiak@samsung.com>

Changes since v1:
Fix show-gadgets example: get attributes before dereferencing them

Reported-by: Philippe De Swert <philippe.deswert@jollamobile.com>
Signed-off-by: Krzysztof Opasiak <k.opasiak@samsung.com>

kopasiak committed:
libusbg: fix: Remove unused variables

Signed-off-by: Krzysztof Opasiak <k.opasiak@samsung.com>

kopasiak committed:
libusbg: Allow to set all gadget attrs using one function.

Having specialized functions for each attribute may be
sometime inconvenient to use. Provide also function
whihc allows to set attribute selected by parameter.

Signed-off-by: Krzysztof Opasiak <k.opasiak@samsung.com>

kopasiak committed:
libusbg: fix: reverse comparsion to fix error

Signed-off-by: Krzysztof Opasiak <k.opasiak@samsung.com>

kopasiak committed:
libusbg: Return suitable error codes instead of -1

Signed-off-by: Krzysztof Opasiak <k.opasiak@samsung.com>

kopasiak committed:
libusbg: Make usbg_lookup_function_type() publicly available

Users of library may often receive function type as a string.
Instead of implementing parsing of such string in each program
it would be more convenient to provide such support in library.

Signed-off-by: Krzysztof Opasiak <k.opasiak@samsung.com>

kopasiak committed:
libusbg: Rework API to use usbg_udc structure

Using string as udc identifier provides a lot
of troubles. To be more consistent with rest of
API rework it to start using usbg_udc structure
instead of using char *.

Signed-off-by: Krzysztof Opasiak <k.opasiak@samsung.com>

kopasiak commented:
Please don't merge it now. Not all comments are fixed here. I will send update shortly

kopasiak committed:
libusbg: Add own name to AUTHORS file.

Signed-off-by: Krzysztof Opasiak <k.opasiak@samsung.com>

kopasiak committed:
libusbg: Don't print errno when it has not been directly set

Replace ERRORNO() macro with ERROR(). Difference betwen
them is that ERRORNO() prints also standard error string
based on errno value. This variable should not be accessed
when called function doesn't directly set it.

Signed-off-by: Krzysztof Opasiak <k.opasiak@samsung.com>

kopasiak committed:
libusbg: Add max and min values to enums

In some cases user of library would like to iterate over
all enum values. Depending on enum values order is not
a good idea so provide suitable min and max values to alow
doing something like:

int i;
for(i = USBG_GADGET_ATTR_MIN; i < USBG_GADGET_ATTR_MAX; ++i)
	do_something_useful(i);

Moreover, ensure that function_names and gadget_attr_names
arrays has allways the same size as number of meaningful
values in related enums.

Signed-off-by: Krzysztof Opasiak <k.opasiak@samsung.com>

kopasiak committed:
libusbg: Add udc structure

Add structure to store informations about available udcs
instead of using their names as a string.

Signed-off-by: Krzysztof Opasiak <k.opasiak@samsung.com>
